src/st/cat/unk_3872C.c

```diff
@@ -70,16 +94,63 @@ void func_us_801B87E8(Entity* self) {
             }
             prim->x0 = prim->x2 = base_x + (2 * ((cntr & 7) * 31));
             prim->x1 = prim->x3 = prim->x0 + 0x3E;
-            prim = prim->next;
             cntr++;
+            prim = prim->next;
         }
-    }
 
-    castleFlag = g_CastleFlags[CAT_SPIKE_ROOM_LIT];
-    do {
-        self->ext.et_801B87E8.unk80 = castleFlag;
-        FntPrint("base_x:%04x\n", base_x);
-    } while (0);
+        self->ext.et_801B87E8.unk80 = g_CastleFlags[CAT_SPIKE_ROOM_LIT];
+    }
+    FntPrint("base_x:%04x\n", base_x);
 }
 
-INCLUDE_ASM("st/cat/nonmatchings/unk_3872C", func_us_801B8AD0);
+void func_us_801B8AD0(Entity* self) {
+    FntPrint("col step %x\n", self->step);
+    switch (self->step) {
+    case 0:
+        InitializeEntity(g_EInitSpawner);
+        if (self->params) {
+            g_api.func_800EA5E4(0x800B);
+            g_Tilemap.flags &= 0xFF7F;
+            DestroyEntity(self);
+            break;
+        }
+
+        if (g_CastleFlags[CAT_SPIKE_ROOM_LIT]) {
+            g_Tilemap.flags |= LAYER_SEMI_TRANS;
+            g_api.func_800EA5E4(0x8004);
+            g_api.func_800EA5E4(0x8006);
+            g_api.func_800EA5E4(0x8007);
+            g_api.func_800EA5E4(0x8008);
+            g_api.func_800EA5E4(0x800B);
```

Comment:
I am pretty sure these are palettes
https://github.com/Xeeynamo/sotn-decomp/blob/ee36845b96a7e8cefa0f441575d9700178fc2346/src/dra/4A538.c#L35-L47

And these should be eg. `PAL_FLAG(0xB)`

But other places in the code are using `ANIMSET_OVL` in `func_800EA5E4` invocations, so I don't feel confident making that change here

Reply:
I agree with you, I'm pretty sure those are palettes also.  That function is pretty heavily clut based and D_800A3BB8 is also used for a clut value.

Reply:
I think the primary issue is the fact that the function definition itself uses that argument as a clut value.  If it is in fact an animset, then the function needs to be adjusted.